app/code/core/Mage/Adminhtml/Block/Catalog/Category/Tab/Product.php

```diff
@@ -142,6 +143,34 @@ protected function _prepareColumns()
             'editable'  => !$this->getCategory()->getProductsReadonly()
             //'renderer'  => 'adminhtml/widget_grid_column_renderer_input'
         ]);
+        $this->addColumn('status', [
+            'header'    => Mage::helper('catalog')->__('Status'),
+            'width'     => '70',
+            'index'     => 'status',
+            'type'      => 'options',
+            'options' => Mage::getSingleton('catalog/product_status')->getOptionArray(),
+        ]);
+        $this->addColumn('action', [
+            'header'    => Mage::helper('catalog')->__('Action'),
+            'width'     => '50px',
+            'type'      => 'action',
+            'getter'    => 'getId',
+            'actions'   => [
+                [
+                    'caption'   => Mage::helper('catalog')->__('Edit'),
+                    'id'        => 'editlink',
+                    'field'     => 'id',
+                    'onclick'  => 'event.stopPropagation()',
+                    'url'       => [
+                        'base'      => 'adminhtml/catalog_product/edit',
+                        'params'    => ['store' => $this->getRequest()->getParam('store')],
+                    ]
+                ],
+            ],
+            'filter' => false,
+            'sortable' => false,
+            'index' => 'stores',
+        ]);
 
```

Comment:
I didn't want to edit the file and create a commit. This change makes the grid the same with the ones in **Related**, **Up-sells** and **Cross-sells**. 

For the **Price** column the width was set to 1 ```'width'     => '1',```. It should be removed as I did bellow.

```php
protected function _prepareColumns()
    {
        if (!$this->getCategory()->getProductsReadonly()) {
            $this->addColumn('in_category', [
                'header_css_class' => 'a-center',
                'type'      => 'checkbox',
                'name'      => 'in_category',
                'values'    => $this->_getSelectedProducts(),
                'align'     => 'center',
                'index'     => 'entity_id'
            ]);
        }
        $this->addColumn('entity_id', [
            'header'    => Mage::helper('catalog')->__('ID'),
            'sortable'  => true,
            'width'     => '60',
            'index'     => 'entity_id'
        ]);
        $this->addColumn('name', [
            'header'    => Mage::helper('catalog')->__('Name'),
            'index'     => 'name'
        ]);

        $this->addColumn('type', [
            'header'    => Mage::helper('catalog')->__('Type'),
            'width'     => 100,
            'index'     => 'type_id',
            'type'      => 'options',
            'options'   => Mage::getSingleton('catalog/product_type')->getOptionArray(),
        ]);

        $sets = Mage::getResourceModel('eav/entity_attribute_set_collection')
            ->setEntityTypeFilter(Mage::getModel('catalog/product')->getResource()->getTypeId())
            ->load()
            ->toOptionHash();

        $this->addColumn('set_name', [
            'header'    => Mage::helper('catalog')->__('Attrib. Set Name'),
            'width'     => 130,
            'index'     => 'attribute_set_id',
            'type'      => 'options',
            'options'   => $sets,
        ]);

        $this->addColumn('status', [
            'header'    => Mage::helper('catalog')->__('Status'),
            'width'     => 90,
            'index'     => 'status',
            'type'      => 'options',
            'options'   => Mage::getSingleton('catalog/product_status')->getOptionArray(),
        ]);

        $this->addColumn('visibility', [
            'header'    => Mage::helper('catalog')->__('Visibility'),
            'width'     => 90,
            'index'     => 'visibility',
            'type'      => 'options',
            'options'   => Mage::getSingleton('catalog/product_visibility')->getOptionArray(),
        ]);


        $this->addColumn('sku', [
            'header'    => Mage::helper('catalog')->__('SKU'),
            'width'     => '80',
            'index'     => 'sku'
        ]);
        $this->addColumn('price', [
            'header'    => Mage::helper('catalog')->__('Price'),
            'type'  => 'currency',
            'currency_code' => (string) Mage::getStoreConfig(Mage_Directory_Model_Currency::XML_PATH_CURRENCY_BASE),
            'index'     => 'price'
        ]);
        $this->addColumn('position', [
            'header'    => Mage::helper('catalog')->__('Position'),
            'width'     => '1',
            'type'      => 'number',
            'index'     => 'position',
            'editable'  => !$this->getCategory()->getProductsReadonly()
            //'renderer'  => 'adminhtml/widget_grid_column_renderer_input'
        ]);
        $this->addColumn('action', [
            'header'    => Mage::helper('catalog')->__('Action'),
            'width'     => '50px',
            'type'      => 'action',
            'getter'    => 'getId',
            'actions'   => [
                [
                    'caption'   => Mage::helper('catalog')->__('Edit'),
                    'id'        => 'editlink',
                    'field'     => 'id',
                    'onclick'  => 'event.stopPropagation()',
                    'url'       => [
                        'base'      => 'adminhtml/catalog_product/edit',
                        'params'    => ['store' => $this->getRequest()->getParam('store')],
                    ]
                ],
            ],
            'filter' => false,
            'sortable' => false,
            'index' => 'stores',
        ]);

        return parent::_prepareColumns();
    }
```

The approach for the **Edit** action will be discussed later.